### PR TITLE
Add addPrefixToTests config

### PR DESCRIPTION
**Pull request by joaomqcunha**
**Motivation:** I'm currently using a protractor config with `multicapablities` for different screen dimensions (S, M, L, etc...) and currently the `protractor-screenshoter-plugin` does not differentiate the test description between each of those results, making it harder to understand which was the screen dimension that started failing.

**Changes:**
* Added a new config called `addPrefixToTests`;
* Added the relevant automated tests;
* Updated the unit-tests;

### CODE REVIEW

**Comment by joaomqcunha** [comment]
@azachar I have some tests that are currently failing but apparently they are not related with this last developments. I've tried to run the test against the current version in the `master` branch and they are also breaking. Can you please check if this is in-fact a problem in this new branch?

**Comment by azachar** [comment]
Hi there,

Sure I have a look and let you known.

I guess that the master tests are failing, I try to fix them.


You know they are very fragile since they depend on angularjs domain as an example server to run protractor tests against. This needs to be changed in order to have master tests stable. Is here any volunteer to create an infrastructure for it


Cheers,
Andrej

**Comment by joaomqcunha** [comment]
That change will probably require a completely refactor on the tests because you will have to create the an all new battery using a dummy application being served locally, right? And in my opinion it doesn't worth it because the "AngularJS" site will not go down soon and also... that is probably not the root cause for the tests are failing in master. Let me know which were your findings 👍

**Comment by azachar** [comment]
Dear @joaomqcunha,

I finally manage to fix the build process. So could you please rebase and squash your solution on the latest `master` branch?

Thanks!

Cheers,
Andrej

**Comment by joaomqcunha** [comment]
Thanks @azachar , this is now working! Feel free to accept it and generate a new version.

File: index.js [hunk 1]
```diff
@@ -436,19 +437,31 @@ protractorUtil.registerJasmineReporter = function(context) {
...
+        protractorUtil.test = {
+          start: moment(),
+          specScreenshots: [],
+          specLogs: [],
+          specHtmls: [],
+          failedExpectations: [],
+          passedExpectations: [],
+          prefix: ''
+        };
+        global.browser.getProcessedConfig().then(function(config) {
```
Comment (azachar) [inline index.js]: I think it missing here a return statement

Comment (joaocunha-kununu) [reply index.js]: `specStarted` doesn't need a return value. Am I right? Therefore, I don't think we need a return statement.

**Comment by joaomqcunha** [comment]
@azachar Can we merge this code into master and generate a minor/patch version? I'm waiting for this work for a while now. Let me know your thoughts, thanks.

**Pull request merged**